protocol-units/bridge/sources/METHBridge.move

```diff
@@ -0,0 +1,455 @@
+    use aptos_framework::event::{EventHandle};
+    use aptos_framework::account;
+
+    const BRIDGE_ACCOUNT: address = @0x1; //Should change this later to actual bridge address
+    const EPENDING_WITHDRAWAL_NOT_FOUND: u64 = 3;
+
+    struct BridgeAccount has key {
+        deposit_events: EventHandle<DepositEvent>,
+        pending_withdrawal_events: EventHandle<PendingWithdrawalEvent>,
+        pending_withdrawals: vector<PendingWithdrawal>,
+        nonce: u256,
+    }
+
+    struct Deposit has drop, store {
+        owner: address,
+        token_id: u128,
+        nonce: u256,
+        amount: u64,
+    }
+
+    struct DepositEvent has drop, store {
+        deposit: Deposit,
+    }
+
+    struct PendingWithdrawalRequest has drop, store {
+        owner: address,
+        token_id: u128,
+        amount: u64,
+    }
+
+    struct PendingWithdrawal has drop, store {
+        request: PendingWithdrawalRequest,
+        nonce: u256,
+    }
+
+    struct PendingWithdrawalEvent has drop, store {
+        nonce: u256,
+        owner: address,
+        token_id: u128,
+        amount: u64,
+    }
+
+    public entry fun deposit(
+        trusted: &signer,
+        deposit_owner: address,
+        token_id: u128,
+        nonce: u256,
+        amount: u64
+    ) acquires BridgeAccount {
+        let trusted_address = signer::address_of(trusted);
+        assert!(trusted_address == BRIDGE_ACCOUNT, 1); // Verify trusted signer
+
+        let bridge_account = borrow_global_mut<BridgeAccount>(trusted_address);
+        let coin = coin::withdraw<AptosCoin>(trusted, amount);
+        coin::deposit(deposit_owner, coin);
+
+        let deposit = Deposit {
+            owner: deposit_owner,
+            token_id,
+            nonce,
+            amount,
+        };
+        event::emit_event(
+            &mut bridge_account.deposit_events,
+            DepositEvent { deposit }
+        );
+    }
+
+    public entry fun withdraw(
+        owner: &signer,
+        token_id: u128,
+        amount: u64
+    ) acquires BridgeAccount {
+        let owner_address = signer::address_of(owner);
+        let coin = coin::withdraw<AptosCoin>(owner, amount);
+        let bridge_account = borrow_global_mut<BridgeAccount>(BRIDGE_ACCOUNT);
+        coin::deposit(BRIDGE_ACCOUNT, coin);
+        let nonce = bridge_account.nonce;
+        bridge_account.nonce = nonce + 1;
+
+        let request = PendingWithdrawalRequest {
+            owner: owner_address,
+            token_id,
+            amount,
+        };
+        let pending_withdrawal = PendingWithdrawal {
+            request,
+            nonce,
+        };
+
+        let pending_withdrawal_event = PendingWithdrawalEvent {
+            nonce,
+            owner: owner_address,
+            token_id,
+            amount,
+        };
+
+        // Add the pending withdrawal to the pending_withdrawals vector
+        vector::push_back(&mut bridge_account.pending_withdrawals, pending_withdrawal);
```

Comment:
I think this might have been mentioned but couldn't we use a different type of container for this for the lookups?